index.js

```diff
@@ -132,6 +132,7 @@ function format(/** @type {string} */ html, indent = "  ", width = 80) {
       if (tagName == specialElement) specialElement = null;
       if (!specialElement) {
         --level;
```

Comment:
Move the check here instead:

```suggestion
        if (level > 0) --level;
```